### PR TITLE
Add documentation on error summary

### PR DESCRIPTION
**Pull request by saarasat**
Add documentation on validation error summary.

### CODE REVIEW

File: _patterns/error-summary/basic.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+layout: pattern
+---
+
+<div class="error-summary">
+	<div class="error-summary__text">
+		<h4>Submission failed</h4>
+		<p>Check your answers before trying again</p>
+		<p>Label of the field: error message from the system</p>
```
Comment (mtthwhggns) [inline _patterns/error-summary/basic.html]: Have we considered displaying these in a `<ul>`, as there is potentially a list of them?

File: components/error-summary.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+---
+layout: component-detail
+group: components
+
+title: Error summary
+description: Error summary is used to present validation errors of a form.
+status: Beta
+
+github_link: https://github.com/apluslms/a-plus-design-system/issues/9
```
Comment (mtthwhggns) [inline components/error-summary.md]: This URL should point to an issue, as the PR will be closed. You could create an empty issue for further discussions about the component, as I have done for the previous components.

File: _patterns/error-summary/basic.html (hunk 1 above)
Comment (mtthwhggns) [inline _patterns/error-summary/basic.html]: Should the label of the field be distinct somehow - maybe `<strong>`?

File: components/error-summary.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  pattern: error-summary/basic.html
+---
+
+## General guidelines
+
+- You should always present the status of the submission or alternatively a heading indicating that there was a problem. For exercises the heading "Submission failed" has been generally used.  
+
+- Provide the name of the field with the error and preferably a link to the erroneous field. Add the error message from the system after the name of the field.
+
+- Make keyboard focus to the error summary.
```
Comment (mtthwhggns) [inline components/error-summary.md]:
```suggestion
- Move keyboard focus to the error summary.
```

File: components/error-summary.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+---
+layout: component-detail
+group: components
+
+title: Error summary
+description: Error summary is used to present validation errors of a form.
```
Comment (mtthwhggns) [inline components/error-summary.md]:
```suggestion
description: An error summary is used to present validation errors of a form.
```
Just to match the others on the "Overview" page

**Review by mtthwhggns — COMMENTED** [review]
Hi @saarasat 
I've checked this out and tested it, and it looks pretty good.

I've added a couple of tiny language things and a couple of tiny component things, but overall this seems pretty good. 

Once these have been checked, I think it's ready-to-merge.

File: components/error-summary.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+- Make your error messages clear and concise. 
+
+## Known issues
+
+- The current error summary does not provide links to the specific fields which have errors. 
+- There is no consistent heading text yet for other types of forms than exercise forms.
+
+## Location of the error summary
+
+Use at the top of an view. In most cases, a view is one web page, but there might be some exceptions:
```
Comment (mtthwhggns) [inline components/error-summary.md]:
```suggestion
Use at the top of a view. In most cases, a view is one web page, but there might be some exceptions:
```
Sorry I missed this in my actual review!

File: _patterns/error-summary/basic.html (hunk 1 above)
Comment (saarasat) [reply _patterns/error-summary/basic.html]: I now made all the changes here, they were all valid points. And you are correct, `ul` is probably the right way to go. However, I don't like the bullet, I guess that could be removed then in the aplus sass-files with a separate small pr?

File: components/error-summary.md (hunk 5 above)
Comment (saarasat) [reply components/error-summary.md]: Thanks for pointing out! this has also been changed now.

File: _patterns/error-summary/basic.html (hunk 1 above)
Comment (raphendyr) [reply _patterns/error-summary/basic.html]: Yes. You can change the styles to remove the list look.

**Comment by markkuriekkinen** [comment]
This PR seems to be ready and complete, or is there still something under construction? I could merge this.

**Comment by saarasat** [comment]
This should be ready :) I am going to create the small pr for those bullets in A+ today

**Comment by mtthwhggns** [comment]
> This should be ready :) I am going to create the small pr for those bullets in A+ today

Since the CSS file here will need updating after that, maybe we should keep this PR open until after that?

File: _patterns/error-summary/basic.html (hunk 1 above)
Comment (saarasat) [reply _patterns/error-summary/basic.html]: PR for this has now been created: https://github.com/apluslms/a-plus/pull/658

**Comment by markkuriekkinen** [comment]
> > This should be ready :) I am going to create the small pr for those bullets in A+ today
> 
> Since the CSS file here will need updating after that, maybe we should keep this PR open until after that?

I'm not yet familiar with how the design-system is implemented. Do we need to copy CSS from a-plus to this design-system so that the examples in the design-system are rendered in the same style as in a-plus, or what is the idea? What kind of CSS needs changes in the design-system?

**Comment by mtthwhggns** [comment]
> I'm not yet familiar with how the design-system is implemented. Do we need to copy CSS from a-plus to this design-system so that the examples in the design-system are rendered in the same style as in a-plus, or what is the idea? What kind of CSS needs changes in the design-system?

For now, CSS changes in `aplus` cause `main.css` to be regenerated using SASS. This single file is then manually copied here.

I know this is not ideal, but given we aren't expecting to update this too often, doing it this way seems pretty painless until we can get a proper pipeline going with the assets.

**Pull request merged**